First draft of Bednets (#7)
mrc-ide/malsimtutorial: inst/tutorials/bednets/bednets.Rmd

---
title: "Bednets"
output: learnr::tutorial
runtime: shiny_prerendered
---

```{r setup, include=FALSE}
library(learnr)
library(malariasimulation)
library(dplyr)
library(tidyr)
library(ggplot2)
knitr::opts_chunk$set(echo = FALSE)
```

## Introduction

> Another widely used intervention that can be simulated using `malariasimulation` is the distribution of insecticidal bed nets (ITNs/LLINs). `malariasimulation` allows the users to tune a number of important components of both the bed nets to be distributed (e.g. efficacy, longevity) and the way in which they are distriubted (e.g. timing, coverage, and target population). In this tutorial, we will build on what we have learnt so far to learn how to simulate bed net distribution campaigns.

Reviewer: ```suggestion
Another widely used intervention that can be simulated using `malariasimulation` is the distribution of insecticide treated bed nets (ITNs/LLINs). `malariasimulation` allows the users to tune a number of important components of both the bed nets to be distributed (e.g. efficacy, longevity) and the way in which they are distributed (e.g. timing, coverage, and target population). In this tutorial, we will build on what we have learnt so far to learn how to simulate bed net distribution campaigns.
```

Author: Changes implemented as requested!

# Load the base parameter set:
parameters <- get_parameters(overrides = list(human_population = 1000))

# Load the bed net parameters into the parameter list using set_bednets()
parameters <- set_bednets(parameters = parameters,
                          timesteps = c(1, 4) * 365,
                          coverages = c(0.5, 0.5),
                          retention = 5 * 365,
                          dn0 = matrix(c(0.533, 0.533), nrow = 2, ncol = 1),
                          rn =  matrix(c(0.56, 0.56), nrow = 2, ncol = 1),
                          rnm =  matrix(c(0.24, 0.24), nrow = 2, ncol = 1), 
                          gamman =  rep(2.64 * 365, 2))

```

Here, we have specified a bed net campaign involving two distribution events, one to occur after 1 year and another after 4 years, which will achieve a coverage of 50%. We are assuming that individuals will, on average, retain a bed net for 5 years (`retention`), and that each bed net has a half life (`gamman`) of \~960 days.

```{r quiz-1}
quiz(caption = "Parameters Qs",
     question("What mosquito death rate (dn0) have we specified in the chunk above?",
              answer("365"),

>               answer("0.56", correct = TRUE),

Reviewer 2: 0.56 is rn, `dn0` is 0.533

Author: Good spot - fixed!

              answer("0.24"),
              answer("0"), allow_retry = T),
     question("How many columns must the matrix inputs for dn0, rn, and rnm contain if we specify bed net events to be updated 5 times (i.e. we want parameters to be updated on 5 timesteps)?",
              answer("3"),

>               answer("Depends on the number of mosquito species"),

Reviewer 2: ```suggestion
              answer("It depends on the number of mosquito species"),
```

Author: Changed!

                          retention = 5 * 365,
                          dn0 = matrix(c(0.533, 0.533), nrow = 2, ncol = 1),
                          rn =  matrix(c(0.56, 0.56), nrow = 2, ncol = 1),
                          rnm =  matrix(c(0.24, 0.24), nrow = 2, ncol = 1), 
                          gamman =  rep(2.64 * 365, 2))

# Equilibrate the model parameters to an initial EIR:
parameters <- set_equilibrium(parameters = parameters, 
                              init_EIR = 8)

# Run the simulation
simulation_output <- run_simulation(timesteps = 7 * 365, 
                                    parameters = parameters)

# Plot the mosquito population through time:
simulation_output |>
  mutate(total_mosquito_population = Sm_gamb_count + Im_gamb_count + Pm_gamb_count) |>
  ggplot(aes(x = timestep, y = total_mosquito_population)) +
  geom_line(linewidth = 1.2, colour = "deepskyblue2") +
  theme_bw() +
  labs(x = "Time (days)", y = "Adult Female Mosquito Population Size") +
  scale_y_continuous(limits = c(0, 7500)) +
  geom_vline(xintercept = c(365, 1460), linetype = "dashed")

# Plot the mosquito population through time:
simulation_output |>

>   mutate(pfpr210 = n_detect_lm_730_3650 / n_730_3650) |>

Reviewer 2: This is a `dev` malariasimulation branch naming convention. I think we should stick with the main branch, which would make it `n_detect_730_3650` I think

Author: Changed!

## Specifying a campaign when simulating multiple mosquito species

The default behaviour is for `malariasimulation` to simulate a single mosquito species, *Anopheles gambiae*. However, as we've discussed, `malariasimulation` supports the simulation of settings with multiple mosquito species, and has in-built parameter sets for *An. funestus* (`fun_params`), *An. arabiensis* (`arab_params`), and *An. stephensi* (`steph_params`). In this section, we'll see how to set up the same bed net campaign we've just used, but with two vector species.

We won't focus on the parameters in these sets today (although feel free to ask later), but we can view the parameter sets:

```{r set_species_demo_1, exercise = TRUE}

# View the parameter set for An. funestus:
fun_params

# ... and view the associated helper files:
?fun_params

```

> Similar to `set_drugs()` in the treatment exercise, the `set_species()` function is used to load the mosquito species parameters of interest into our list of `malariasimulation` parameters. The `set_species()` function has three arguments:

Reviewer 2: Is it possible that attendess may have jumped here before seeing `set_drugs()`
```suggestion
The `set_species()` function is used to load the mosquito species parameters of interest into our list of `malariasimulation` parameters. The `set_species()` function has three arguments:
```

Author: Yeah quite possible - have made the change as suggested!
